device.go

```diff
@@ -96,6 +97,50 @@ func (d *device) SwapPartitions() error {
 	return nil
 }
 
+// chunkedCopy copies data from in to out in chunks of exactly chunkSize
+// bytes.
+// Data is held in memory until chunkSize bytes are available to be written.
+func chunkedCopy(out io.Writer, in io.Reader, chunkSize int64) (total_written int64, err error) {
+	buf := bytes.NewBuffer(make([]byte, 0, chunkSize))
+
+	total_written = 0
+
+	for {
+		buf.Reset()
+
+		// read chunkSize bytes into buf
+		bytesRead, readErr := io.CopyN(buf, in, chunkSize)
+
+		if bytesRead > 0 {
+			// write all of buf to out
+			bytesWritten, writeErr := buf.WriteTo(out)
+
+			total_written += bytesWritten
+
+			if writeErr != nil {
+				return total_written, writeErr
+			}
+
+			if bytesWritten != bytesRead {
+				panic(fmt.Sprintf(
+					"Unexpected short write: attempted %v bytes but only wrote %v",
+					bytesRead,
+					bytesWritten,
+				))
```

Comment:
Should return error here, not panic.

Reply:
Updated `.travis.yml` in subsequent commit.